### PR TITLE
UCP: fix stup ep lookup in error handler

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
Fix error handling part of #1578

### CODE REVIEW

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -300,6 +304,9 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
     if (ucp_ep_errh_iter != kh_end(&worker->ep_errh_hash)) {
         err_handler = kh_val(&worker->ep_errh_hash, ucp_ep_errh_iter);
         err_handler.cb(err_handler.arg, ucp_ep, status);
+    } else {
+        ucs_error("Error %s was not handled for ep %p",
```
Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: i don't think we should add this print, let the transport print the error as today

File: src/ucp/core/ucp_worker.c [hunk 2]
```diff
@@ -248,7 +249,10 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
     /* TODO: need to optimize uct_ep -> ucp_ep lookup */
     kh_foreach(&worker->ep_hash, dest_uuid, ucp_ep_iter, {
         for (lane = 0; lane < ucp_ep_num_lanes(ucp_ep_iter); ++lane) {
-            if (uct_ep == ucp_ep_iter->uct_eps[lane]) {
+            stub_ep = ucp_stub_ep_test(ucp_ep_iter->uct_eps[lane]) ?
```
Comment (yosefe) [inline src/ucp/core/ucp_worker.c]:
```
   aux_ep = ucp_stub_ep_test(ucp_ep_iter->uct_eps[lane]) ?
            ucs_derived_of(ucp_ep_iter->uct_eps[lane], ucp_stub_ep_t)->aux_ep : NULL;
   if ((uct_ep == ucp_ep_iter->uct_eps[lane]) || (uct_ep == aux_ep)) {
```

File: src/ucp/core/ucp_worker.c (hunk 1 above)
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: The transport calls error handling callback from UCP

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1829/ for details.

File: src/ucp/core/ucp_worker.c (hunk 1 above)
Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: but we have this in the transport for example:
```
    uct_ib_mlx5_completion_with_err(arg, UCS_LOG_LEVEL_ERROR); <----
    uct_dc_handle_failure(ib_iface, qp_num);
``
isn't this enough?

Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: This is a bug :) I wouldn't do error print on handled error. Probably we need to pass some extra info to upper level for debug purposes.

Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: as i see it there are 2 options:
1. pass extra info to upper level
2. return status to lower level and let it print the error if unhandled
either way is good, as long as the error is printed only if unhandled, no more than once, and contains all relevant info.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3769/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c (hunk 1 above)
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: Agree, added to https://github.com/openucx/ucx/issues/755

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1842/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3783/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3790/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlnx:retest

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3803/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3805/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3812/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3815/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3817/ for details (Mellanox internal link).

**Pull request merged**